### PR TITLE
implement login and signup

### PR DESCRIPTION
**Pull request by vinayluzrao**

### CODE REVIEW

File: app/src/main/java/com/rjt/property_management/extension_functions/Context.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.rjt.property_management.extension_functions
+
+import android.content.Context
+import android.content.Context.MODE_PRIVATE
+import com.rjt.property_management.ApiService
+import com.rjt.property_management.models.Landlord
+
+
+fun Context.storeSessionInformation(info: ApiService.SuccessfulLoginResponse) {
```
Comment (samuelmoutah) [inline app/src/main/java/com/rjt/property_management/extension_functions/Context.kt]: I know  Context.storeSessionInformation is function name. but What' Context. ??

Comment (vinayluzrao) [reply app/src/main/java/com/rjt/property_management/extension_functions/Context.kt]: Context is a class that contains information about the application (like shared preferences, which is what I needed for this method). Activities inherit from it. This function is a Kotlin extension function.

File: app/src/main/java/com/rjt/property_management/activities/SignUpActivity.kt [hunk 2]
```diff
@@ -1,13 +1,34 @@
...
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_sign_up)
+        val binding: SignUpActivityBinding = DataBindingUtil.setContentView(
+            this,
+            R.layout.sign_up_activity
+        )
+        binding.lifecycleOwner = this
```
[the remark this answers is not in the document]
Comment (vinayluzrao) [reply app/src/main/java/com/rjt/property_management/activities/SignUpActivity.kt]: We have to set the lifecycleowner when we use live data with data binding. For more details, check out https://developer.android.com/topic/libraries/data-binding/architecture#livedata .

File: app/src/main/java/com/rjt/property_management/ApiService.kt [hunk 3]
```diff
@@ -1,24 +1,73 @@
...
-        fun getProperties() = retrofit.create(ApiService::class.java).getProperties()
     }
 
     @GET("pro_mgt_property_all.php")
     fun getProperties(): Call<AllPropertiesResponse>
 
     data class AllPropertiesResponse(val Property: List<Property>)
+
+    @GET("pro_mgt_login.php")
+    fun login(@Query("email") email: String, @Query("password") password: String):
```
[the remark this answers is not in the document]
Comment (vinayluzrao) [reply app/src/main/java/com/rjt/property_management/ApiService.kt]: it is a retrofit annotation that allows for dynamically setting query strings in URLs. for more information, read this https://square.github.io/retrofit/2.x/retrofit/index.html?retrofit2/http/Query.html

File: app/src/main/java/com/rjt/property_management/ApiService.kt [hunk 4]
```diff
@@ -1,24 +1,73 @@
...
     companion object {
-        private val retrofit =
+        private val serviceWithGsonConverter =
             Retrofit.Builder()
                 .addConverterFactory(GsonConverterFactory.create())
                 .baseUrl("http://rjtmobile.com/aamir/property-mgmt/")
                 .build()
+                .create(ApiService::class.java)
+
+        private val serviceWithoutGsonConverter =
```
[the remark this answers is not in the document]
Comment (vinayluzrao) [reply app/src/main/java/com/rjt/property_management/ApiService.kt]: The "register" API end point returns a string that cannot be parsed as a JSON. GSON can only parse valid JSON. Therefore, trying to use the GSON converter with that end point causes the app to crash.

**Pull request merged**